src/AzureClient/EntryPoint/EntryPointGenerator.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+using Microsoft.Extensions.Logging;
+using Microsoft.Quantum.IQSharp.Common;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.IQSharp.AzureClient
+{
+    /// <inheritdoc/>
+    internal class EntryPointGenerator : IEntryPointGenerator
+    {
+        private ICompilerService Compiler { get; }
+        private ILogger<EntryPointGenerator> Logger { get; }
+        private IWorkspace Workspace { get; }
+        private ISnippets Snippets { get; }
+        public IReferences References { get; }
+        public AssemblyInfo WorkspaceAssemblyInfo { get; set; } = new AssemblyInfo(null);
+        public AssemblyInfo SnippetsAssemblyInfo { get; set; } = new AssemblyInfo(null);
+        public AssemblyInfo EntryPointAssemblyInfo { get; set; } = new AssemblyInfo(null);
+
+        public EntryPointGenerator(
+            ICompilerService compiler,
+            IWorkspace workspace,
+            ISnippets snippets,
+            IReferences references,
+            ILogger<EntryPointGenerator> logger,
+            IEventService eventService)
+        {
+            Compiler = compiler;
+            Workspace = workspace;
+            Snippets = snippets;
+            References = references;
+            Logger = logger;
+
+            AssemblyLoadContext.Default.Resolving += Resolve;
+
+            eventService?.TriggerServiceInitialized<IEntryPointGenerator>(this);
+        }
+        
+        /// <summary>
+        /// Because the assemblies are loaded into memory, we need to provide this method to the AssemblyLoadContext
+        /// such that the Workspace assembly or this assembly is correctly resolved when it is executed for simulation.
+        /// </summary>
+        public Assembly Resolve(AssemblyLoadContext context, AssemblyName name)
+        {
+            if (name.Name == Path.GetFileNameWithoutExtension(EntryPointAssemblyInfo?.Location))
+            {
+                return EntryPointAssemblyInfo.Assembly;
+            }
+            if (name.Name == Path.GetFileNameWithoutExtension(SnippetsAssemblyInfo?.Location))
+            {
+                return SnippetsAssemblyInfo.Assembly;
+            }
+            else if (name.Name == Path.GetFileNameWithoutExtension(WorkspaceAssemblyInfo?.Location))
+            {
+                return WorkspaceAssemblyInfo.Assembly;
+            }
+
+            return null;
+        }
```

Comment:
```suggestion
        public Assembly? Resolve(AssemblyLoadContext context, AssemblyName name) => name.Name switch
        {
            Path.GetFileNameWithoutExtension(EntryPointAssemblyInfo?.Location) => EntryPointAssemblyInfo.Assembly,
            Path.GetFileNameWithoutExtension(SnippetsAssemblyInfo?.Location) => SnippetsAssemblyInfo.Assembly,
            Path.GetFileNameWithoutExtension(WorkspaceAssemblyInfo?.Location) => WorkspaceAssemblyInfo.Assembly,
            _ => null
        };
```

Reply:
Yes, makes sense! I think I need to add some `var s when s ==` in there to make it compile since the match expression is not a constant.